File: intellij-plugin/src/main/java/com/github/tth05/tethintellijplugin/TethLanguage.java
package com.github.tth05.tethintellijplugin;

import com.intellij.lang.Language;

public class TethLanguage extends Language {

    public static final TethLanguage INSTANCE = new TethLanguage();

    private TethLanguage() {
        super("Teth");
    }
}

File: intellij-plugin/src/main/kotlin/com/github/tth05/tethintellijplugin/TethLanguage.kt
package com.github.tth05.tethintellijplugin

import com.github.tth05.tethintellijplugin.TethLanguage
import com.intellij.lang.Language

class TethLanguage : Language("Teth") {

    override fun isCaseSensitive(): Boolean = true

    companion object {
        val INSTANCE: TethLanguage = TethLanguage()
    }
}